packages/ui/src/components/primitives/Message.tsx

```diff
@@ -60,99 +135,216 @@ export const MessageAuthorArea = () => {
 	);
 };
 
-export const MessageContents = () => {
+const LongMessageContents = () => {
 	const { message } = useMessageContext();
 	const { toHTML } = discordMarkdown;
 	const convertedMessageContent = toHTML(message.content);
+	const [messageContentToRender, setMessageContentToRender] = useState<string>(
+		convertedMessageContent,
+	);
+	const [isOpen, setIsOpen] = useState<boolean>(false);
+
+	useEffect(() => {
+		if (isOpen) {
+			setMessageContentToRender(convertedMessageContent);
+		} else {
+			setMessageContentToRender(`${convertedMessageContent.slice(0, 1000)}...`);
+		}
+	}, [convertedMessageContent, isOpen]);
+
 	return (
-		<div
-			className="pt-2 font-body text-ao-black [word-wrap:_break-word] dark:text-ao-white"
-			// The HTML from discord-markdown is escaped
-			dangerouslySetInnerHTML={{
-				__html: convertedMessageContent,
-			}}
-		/>
+		<Collapsible.Root open={isOpen} onOpenChange={setIsOpen}>
+			<div
+				className="pt-2 font-body text-ao-black [word-wrap:_break-word] dark:text-ao-white"
+				// The HTML from discord-markdown is escaped
+				dangerouslySetInnerHTML={{
+					__html: messageContentToRender,
+				}}
+			/>
+			<Collapsible.Trigger asChild>
+				<Button className="mt-2 gap-2" size="sm" variant="outline">
+					{isOpen ? (
+						<>
+							Collapse
+							<ChevronUpIcon className="h-4 w-4" />
+						</>
+					) : (
+						<>
+							Expand
+							<ChevronDownIcon className="h-4 w-4" />
+						</>
+					)}
+				</Button>
+			</Collapsible.Trigger>
+		</Collapsible.Root>
 	);
 };
 
-const MessageModalWrapper = ({
-	children,
-	attachment,
-}: React.PropsWithChildren<{
-	attachment: APIMessageWithDiscordAccount['attachments'][number];
-}>) => {
-	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+export const MessageContents = () => {
+	const { message, collapseContent } = useMessageContext();
+	const { toHTML } = discordMarkdown;
+	const convertedMessageContent = toHTML(message.content);
+
+	if (message.content.length < 1000 || collapseContent === false)
```

Comment:
message.content.length < 1000

Pass in as an optional prop, max message length, with a default, maybe make it a bool | number, if it's set to true, use the default, if its set to a number use that

Reply:
Careful with implemnting this, a number will be typecast to true so make sure to check the type

Reply:
:+1: